src/controllers/controller.h

```diff
@@ -183,6 +187,8 @@ class Controller : public QObject {
     friend class MidiControllerTest;
 };
 
+class Charsets : public QObject {
+};
```

Comment:
?

Reply:
I wanted to expose some common charsets as constants to JS, maybe moving `convertEncoding` to that new object too so that from JS, it would look like: `engine.charsets.convertEncoding(engine.charsets.WELL_KNOWN_CHARSETS.LATIN_9)`, for instance. But I still havent figured out how to do that. I feel like I may be trying to bit off more that I can chew here, provided my C++ skills…

Reply:
I think making it part of the engine class is unnecessary. you can simply add it to the global object, it requires a little bit of boilerplate but not much. See `ControllerScriptEngineBase::initialize()`.

Reply:
Oh you mean have `charsets` exposed in the global namespace? Aye. I could do that.

Reply:
To add an enum, simply add it to the `Charsets` class and make it a `Q_ENUM`. Seems fairly doable IMO, I'm sure you can figure it out. If not, feel free to ask.